File: editor/mode.go
package editor

// Mode represents the current editing mode
type Mode int

const (
	ModeNormal Mode = iota
	ModeInsert
	ModeVisual
)

// String returns the mode name for display
func (m Mode) String() string {
	switch m {
	case ModeNormal:
		return "NORMAL"
	case ModeInsert:
		return "INSERT"
	case ModeVisual:
		return "VISUAL"
	default:
		return "UNKNOWN"
	}
}

File: c_app/src/mode.c
#include "mode.h"

const char *mode_string(Mode m) {
    switch (m) {
    case MODE_NORMAL: return "NORMAL";
    case MODE_INSERT: return "INSERT";
    case MODE_VISUAL: return "VISUAL";
    default:          return "UNKNOWN";
    }
}
